cubzh-file-format-3zh.txt

```diff
@@ -0,0 +1,216 @@
+Cubzh .3zh File Format [10/20/2022]
+
+File
+{
+    Header
+
+    // PNG Preview
+    Chunk 'PREVIEW' : optional
+
+    Chunk 'NB_SHAPES' : N, optional
+
+    // Palette
+    Chunk 'PALETTE' : optional
+
+    // Objects (x N)
+    Chunk 'SHAPE' / Chunk 'OBJECT'
+
+    ...
+
+    Chunk 'SHAPE' / Chunk 'OBJECT'
+}
+-------------------------------------------------------------------------------
+
+
+1. Header
+-------------------------------------------------------------------------------
+# Bytes  | Type       | Value
+-------------------------------------------------------------------------------
+6        | byte       | magic bytes 'CUBZH!' : 'C' 'U' 'B' 'Z' 'H' '!', 'C' is first
+4        | int        | version number : 6
+1        | byte       | p3s compression method : 0 (none), 1 (zip), 2 (count)
```

Comment:
`count` is not a compression method, it just indicated the number of supported compression methods in the enum. We only support 0 (none) or 1 (zip) for now.